specs/agents/configuration.md

```diff
@@ -143,3 +143,48 @@ By default, agents MUST send data to the APM Server at `http://localhost:8200/`.
 If possible, agents SHOULD detect sensible defaults for `service.name` and `service.version`.
 In any case agents MUST include `service.name` - if discovering it is not possible then the default value: `unknown-${service.agent.name}-service` MUST be used.
 This naming pattern allows the UI to display inline help on how to manually configure the name.
+
+### Adding a new configuration option
+
+When adding a new configuration option to the spec, please use the following template:
+
+#### <option_name_in_central_config_notation> configuration
+
+The description of the option.
+Try to phrase it in a way that lets agents easily copy/paste the description into their configuration documentation page.
+This should also match the description in the [central configuration](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts) in the APM UI.
+
+|                |                                                                         |
+|----------------|-------------------------------------------------------------------------|
+| Type           | See the [Configuration Value Types](#configuration-value-types) section |
+| Default        | `<default value>`                                                       |
+| Central config | `<true/false>`                                                          |
+
+Additional description of the option that should not be part of the documentation.
+For example, whether agents SHOULD or MUST implement the option.
+
+#### Adding to central configuration
+
+One of the things to determine when proposing to add a new configuration option is whether it should be added to central configuration.
+In general, we should try to make the majority of options available in central configuration.
+However, sometimes it might not be feasible if agents need to read the value at startup and changing the value at runtime is very difficult to achieve.
+
+When adding a centrally configurable option,
+the spec owner is expected to create a pull request in the Kibana repository to add the option to central configuration.
+As all the configuration options are added declaratively, this is no rocket science.
+See this example PR on how to add an option: https://github.com/elastic/kibana/pull/84678.
+
+Agent devs are not expected to set up a local Kibana development environment.
```

Comment:
Looking at the example PR I linked in this spec (https://github.com/elastic/kibana/pull/84678/files), I'm not sure if it's feasible without a local development environment. It seems like there are some tests that need to be updated and there's a (generated?) [`index.test.ts.snap`](https://github.com/elastic/apm/pull/707/files#diff-6559d63cbbd40d61e56d569ed65255afac02832b9f9b39c4072b6aef382ec253) file that needs to be updated. 

@sqren do you have thoughts on how we could make setting up a dev environment not a requirement?

Reply:
I assume that to make the changes the agent devs will clone Kibana. To run tests and create the snapshots (*.snap files) they'll need Node but they won't need to start Kibana og ES.

Reply:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/testing.md#unit-tests-jest

Reply:
If it helps, I believe these are the steps needed to install, build, and run the APM plugin tests:

```bash
# Prerequisite: Install 'nvm' node version manager tool. This tweaks
# ".bashrc", so you will need to open a new terminal.
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash   # not sure how happy all would be about this, it mucks with profile

git clone git@github.com:elastic/kibana.git  # or git pull
cd kibana

# Install and switch to the exact version of Node that Kibana uses.
nvm install $(cat .nvmrc)

# Install the 'yarn' package manager used by Kibana into this version of node.
npm install -g yarn

# Build Kibana (get a cafe, took ~7m for me.)
time yarn kbn bootstrap

# Run the *primary* agent central config test file.
(cd x-pack; node scripts/jest.js plugins/apm/common/agent_configuration/setting_definitions/index.test.ts)
# There *are* however other tests, so to run all APM tests (~3m for me).
(cd x-pack/plugins/apm; node scripts/test/jest.js)
```

Reply:
I've linked to the contributing guide so that we don't wis this getting stale.